File: solution/322/solution.py
class Solution(object):
    def coinChange(self, coins, amount):
        """
        :type coins: List[int]
        :type amount: int
        :rtype: int
        """
        result = [-1] * (amount + 1)
        result[0] = 0
        for i in xrange(1, amount+1):
            for x in coins:
                if x <= i and result[i-x] != -1:
                    if result[i] == -1 or result[i] > result[i-x] + 1:
                        result[i] = result[i-x] + 1
        return result[amount]

File: solution/322/solution.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        vector<int> result(amount+1, -1);
        result[0] = 0;
        for(int i = 1;i <= amount;i++){
            for(int x=0;x < coins.size();x++){
                int coin = coins[x];
                if(coin<=i&&result[i-coin]!=-1){
                    int num = result[i-coin] + 1;
                    if(result[i] == -1|| result[i] > num){
                        result[i] = num;
                    }
                }
            }
        }
        return result[amount];
    }
};